### PR TITLE
Reuse executable container

### PR DESCRIPTION
**Pull request by g-gaston**
*Description of changes:*
Before this, we were creating a new container for every command and destroying it after running it.
This is a tentative attempt at trying to reuse containers. It will create a long running container in the background during initialization and use the same one for every command.
Once the CLI is done, it will destroy the container and the volumes.

This opens the option to doing things like sharing the same vsphere auth session between multiple `govc` commands.

PD: sorry for the long PR, reviewing commit by commit might help

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jaxesn** [comment]
Each executable will run in its own container vs using 1 container for all command runs?  So all govc commands will run in 1 container and all kubectl commands will run in a different container?

**Comment by g-gaston** [comment]
> Each executable will run in its own container vs using 1 container for all command runs?  So all govc commands will run in 1 container and all kubectl commands will run in a different container?

@jaxesn no, there is one single container for all executables, independently from the binary they run

File: pkg/dependencies/factory_test.go [hunk 1]
```diff
@@ -20,6 +21,11 @@ type factoryTest struct {
 
 func newTest(t *testing.T) *factoryTest {
 	clusterConfigFile := "testdata/cluster_vsphere.yaml"
+	// Disable tools image executable for the tests
+	if err := os.Setenv("MR_TOOLS_DISABLE", "true"); err != nil {
+		t.Fatal(err)
+	}
```
Comment (abhinavmpandey08) [inline pkg/dependencies/factory_test.go:27]: just curious, why is this needed?

File: pkg/executables/builder.go [hunk 2]
```diff
@@ -10,44 +10,50 @@ import (
 	"github.com/aws/eks-anywhere/pkg/logger"
 )
 
-const defaultEksaImage = "public.ecr.aws/l0g8r8j6/eks-anywhere-cli-tools:v1-21-4-ed8ad899e40f0a0e625b7a49d7d90e6077f97a28"
+const defaultEksaImage = "public.ecr.aws/l0g8r8j6/eks-anywhere-cli-tools:v0.1.0-eks-a-v0.0.0-dev-build.529"
```
Comment (abhinavmpandey08) [inline pkg/executables/builder.go:13]: I believe the latest tools image gets pulled in from the bundle. When do we end up using `defaultEksaImage`?

**Review by abhinavmpandey08 — APPROVED** [review]
looks good, just a couple of questions

File: pkg/dependencies/factory_test.go (hunk 1 above)
Comment (g-gaston) [reply pkg/dependencies/factory_test.go:27]: If left enabled, it will try to use the cli tools image, pulling it during init. I don't want unit tests to make network calls

File: pkg/executables/builder.go (hunk 2 above)
Comment (g-gaston) [reply pkg/executables/builder.go:13]: This is used from E2E tests (the test code, not the cli executions) and some of the `eks-a-tool` commands
